src/core/utils/Parser.ts

```diff
@@ -55,6 +55,56 @@ export class Parser {
                 return VNodeType.LINE_BREAK;
         }
     }
+    /**
+     * Return true if the given node is immediately preceding (`side` === 'end')
+     * or following (`side` === 'start') a segment break, to see if its edge
+     * space must be removed.
+     *
+     * @param {Element} node
+     * @param {'start'|'end'} side
+     * @returns {boolean}
+     */
+    static _isAgainstSegmentBreak(node: Element, side: 'start' | 'end'): boolean {
+        const siblingSide = side === 'start' ? 'previousSibling' : 'nextSibling';
+        const sibling = node && (node[siblingSide] as DOMElement);
+        return (sibling && this._isBlock(sibling)) || this._isEdgeOfBlock(node, side);
+    }
+    /**
+     * Return true if the node is a block.
+     * Is considered a block, every node that is displayed as a block, and `BR`
+     * elements.
+     *
+     * @param {DOMElement} node
+     * @returns {boolean}
+     */
+    static _isBlock(node: DOMElement): boolean {
+        const display = window.getComputedStyle(node, null).display;
+        return display === 'block' || node.tagName === 'BR';
+    }
+    /**
+     * Return true if the node is at the given edge of a block.
+     *
+     * @param {Element | Node} node
+     * @param {'start'|'end'} side
+     */
+    static _isEdgeOfBlock(node: Element | Node, side: 'start' | 'end'): boolean {
+        const ancestorsUpToBlock = [];
+
+        // Move up to the first block ancestor
+        let ancestor = node;
+        const isTextNode = ancestor.nodeType === 3;
+        while (ancestor && (isTextNode || !this._isBlock(ancestor as DOMElement))) {
+            ancestorsUpToBlock.push(ancestor);
+            ancestor = ancestor.parentElement;
+        }
```

Comment:
Should be:
```javascript
let isTextNode = ancestor.nodeType === 3;
while (ancestor && (isTextNode || !this._isBlock(ancestor as DOMElement))) {
    ancestorsUpToBlock.push(ancestor);
    ancestor = ancestor.parentElement;
    isTextNode = ancestor.nodeType === 3;
}
```

Reply:
Wel... Then `_isBlock` would be testing more than it should: you can _not_ be a block and still be something else than a text node.